Sources/BrowserServicesKit/SecureVault/SecureVaultDatabaseProvider.swift

```diff
@@ -67,7 +68,14 @@ final class DefaultDatabaseProvider: SecureVaultDatabaseProvider {
         }
 
         let file = try Self.dbFile()
-        db = try DatabaseQueue(path: file.path, configuration: config)
+
+        do {
+            db = try DatabaseQueue(path: file.path, configuration: config)
```

Comment:
Is it possible for DB initialization to fail due to lack of storage space (for temp files)? If that is the case, then removing DB would basically be a bug and data loss for the client. Question is whether there are errors that we can recover from or that are temporary - in such case removing DB may not be the best course of action.

Reply:
Good point, I've added extra steps to validate a new database can be created without dropping the old one: it's renamed to Vault[_N].db.bak after we make sure we can create new database files which should validate we're not dealing with write permissions or write errors and keep users data backup just in case

Reply:
I'd assume there is a finite set of errors we can have when initializing DB - some may be obvious, some may be weird, but generally we should be able to split that set into two paths:
 - Non-recoverable failure - we need to create new DB file as you do here.
 - Recoverable failure - this is product decision what to do with this, we may want to show some dialog to the user (e.g. out of space info).

I don't think we need to solve the latter here, but we should track DB initialization errors (as we do on iOS) and fire error pixel along with error code (and possibly underlying error if available) to understand frequency of this issue and segment it somehow. 

So I suggest:
 - Add error pixels.
 - Create Grafana board for that (I can do this once pixel is there)
 - Schedule a follow up task (2 weeks? maybe before release date?) to validate error types and see if we can improve this solution and make sure we are not breaking something by accident. 

What do you think?